src/app/shared/components/hero/hero.component.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { HeroComponent } from './hero.component';
+import { HeroComponentVM } from './hero.component.vm';
+import { Component, signal } from '@angular/core';
+
+@Component({
+  template: `
+    <app-hero [vm]="vm()">
+      <h1 testH1>The future of irrigation is now</h1>
```

Comment:
Using the the properties as selectors is not correct. Select by tags (h1, p, a, button), select by ids or selext by class. Those are the standards. Adding a custom prop is like hacking, as that prop does not exists in the type definition of the element.